mbuf/mbuf.go

```diff
@@ -179,3 +179,20 @@ func AllocResetAndAppend(p *mempool.Mempool, data *common.CStruct) *Mbuf {
 func (m *Mbuf) GetGoStruct(data *common.CStruct) {
 	C.get_go_struct(mbuf(m), data.Ptr, C.size_t(data.Len))
 }
+
+// HeadRoomSize represents RTE_PKTMBUF_HEADROOM size in concrete mbuf.
+func (m *Mbuf) HeadRoomSize() uint16 {
+	return uint16(C.rte_pktmbuf_headroom(mbuf(m)))
+}
+
+// TailRoomSize represents available length that can be appended to mbuf.
+func (m *Mbuf) TailRoomSize() uint16 {
+	return uint16(C.rte_pktmbuf_tailroom(mbuf(m)))
+}
+
+// DataRoomSize represents mbuf size of HeadRoom, data length and available TailRoom.
+// NOTE: Max available data length that mbuf can hold is DataRoomSize - HeadRoomSize.
+func (m *Mbuf) DataRoomSize() uint16 {
+	rteMbuf := mbuf(m)
+	return m.HeadRoomSize() + uint16(rteMbuf.data_len) + m.TailRoomSize()
```

Comment:
Is the _rte_Mbuf.buf_len_ not suitable in this case?

Reply:
thanks, fixed